via/static/scripts/video_player/hooks/use-stable-callback.ts

```diff
@@ -0,0 +1,23 @@
+import { useRef } from 'preact/hooks';
+
+/**
+ * Return a function which wraps a callback to give it a stable value.
+ *
+ * The wrapper has a stable value across renders, but always forwards to the
+ * callback from the most recent render. This is useful if you want to use a
+ * callback inside a `useEffect` or `useMemo` hook without re-running the effect
+ * or re-computing the memoed value when the callback changes.
+ */
+export function useStableCallback<R, A extends any[], F extends (...a: A) => R>(
+  callback: F
+): F {
+  const wrapper = useRef({
+    callback,
+    call: (...args: A) => wrapper.current.callback(...args),
+  });
+
+  // On each render, save the last callback value.
+  wrapper.current.callback = callback;
+
+  return wrapper.current.call as F;
```

Comment:
So, if I'm getting this right, with this approach you make sure the reference to `call` never changes (hence, it's stable when it comes to side effects), while it forwards calls to `callback`, which is the one changing on every render to make sure its up to date.

Is that correct?

Reply:
That's correct.